test/test_tensor.py

```diff
@@ -239,10 +239,10 @@ def test_tensor_ndarray_dtype(self):
     assert Tensor(arr, dtype=dtypes.float64).dtype == dtypes.float64 # check that it works for something else
 
   def test_tensor_list_dtype(self):
-    arr = [1]
-    assert Tensor(arr).dtype == dtypes.int32
-    assert Tensor(arr, dtype=dtypes.float32).dtype == dtypes.float32
-    assert Tensor(arr, dtype=dtypes.float64).dtype == dtypes.float64
+    for arr in ([1], [[1,1],[1,1]], [[[1,1],[1,1]],[[1,1],[1,1]]]):
```

Comment:
what's the behavior if it's `[[1,1,1], [1,1]]` or `[[1,1,1],[[1,1,1]]]`. add a few more test cases that this can fail